osu.Game.Rulesets.Catch/Objects/Drawables/CaughtObjectContainer.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            randomSeed--;
+
+            lifetimeManager.RemoveEntry(entry);
+            removeDrawable(entry);
+
+            aliveStackedObjects.Remove(entry);
+
+            if (!dropEntryMap.TryGetValue(entry, out var dropEntry)) return;
+
+            dropEntryMap.Remove(entry);
+
+            lifetimeManager.RemoveEntry(dropEntry);
+            removeDrawable(dropEntry);
+        }
+
+        public Vector2 GetPositionInStack(Vector2 position, float displayRadius)
+        {
+            const float radius_div_2 = CatchHitObject.OBJECT_RADIUS / 2;
+            const float allowance = 10;
+
+            int iteration = 0;
+
+            while (aliveStackedObjects.Any(f => Vector2Extensions.Distance(f.PositionInStack, position) < (displayRadius + radius_div_2) / (allowance / 2)))
+            {
+                float diff = (displayRadius + radius_div_2) / allowance;
+
+                position.X += (StatelessRNG.NextSingle(randomSeed, iteration + 2) - 0.5f) * diff * 2;
+                position.Y -= StatelessRNG.NextSingle(randomSeed, iteration * 2 + 1) * diff;
+
+                iteration++;
+            }
+
+            position.X = Math.Clamp(position.X, -CatcherArea.CATCHER_SIZE / 2, CatcherArea.CATCHER_SIZE / 2);
+
+            return position;
+        }
+
+        public Vector2 GetCurrentDropPosition(Vector2 positionInStack)
+        {
+            return StackedObjectContainer.ToSpaceOfOtherDrawable(positionInStack, droppedObjectTarget);
+        }
+
+        public void DropStackedObjects(Action<DrawableCaughtObject> applyTransforms, int mirrorDirection)
+        {
+            dropStackedObjects(Clock.CurrentTime, applyTransforms, mirrorDirection);
+        }
+
+        protected override bool CheckChildrenLife()
+        {
+            bool aliveChanged = base.CheckChildrenLife();
+            aliveChanged |= lifetimeManager.Update(Time.Current);
+            return aliveChanged;
+        }
+
+        private void entryBecameAlive(LifetimeEntry lifetimeEntry)
+        {
+            var entry = (CaughtObjectEntry)lifetimeEntry;
+
+            if (entry.State == CaughtObjectState.Stacked)
+                aliveStackedObjects.Add(entry);
+
+            var drawable = getPooledDrawable(entry.HitObject);
+            drawable.Apply(entry);
+
+            addDrawable(entry, drawable);
+        }
+
+        private void entryBecameDead(LifetimeEntry lifetimeEntry)
+        {
+            var entry = (CaughtObjectEntry)lifetimeEntry;
+
+            if (entry.State == CaughtObjectState.Stacked)
+                aliveStackedObjects.Remove(entry);
+
+            removeDrawable(entry);
+        }
+
+        private void dropStackedObjects(double time, Action<DrawableCaughtObject> applyTransforms, int mirrorDirection)
+        {
+            foreach (var entry in aliveStackedObjects)
+            {
+                entry.LifetimeEnd = time;
+
+                if (!dropEntryMap.TryGetValue(entry, out var dropEntry)) continue;
```

Comment:
This seems like it should be an assertion rather than a soft check. Can this realistically happen and not be a bug?